llvm/test/Transforms/SandboxVectorizer/no_implicit_float.ll

```diff
@@ -0,0 +1,9 @@
+; RUN: opt -passes=sandbox-vectorizer -debug %s 2>&1 | FileCheck %s
+; REQUIRES: asserts
+; Please note that this won't update automatically with update_test_checks.py !
+
+; Check that we early return if the function has the NoImplicitFloat attribute.
+define void @no_implicit_float() noimplicitfloat {
+; CHECK: SBVec: NoImplicitFloat attribute, return.
```

Comment:
these should probably use `update_test_checks.py`, checking debug statements is not recommended since it doesn't necessarily check the actual behavior, plus it only works with assert builds

that would require changing this to IR that would certainly be otherwise vectorized. but given that we don't actually have any vectorization in tree yet, it's kinda a pointless check (i.e. this change is not really meaningfully testable right now), but I'm fine with keeping adding this test with some otherwise vectorizable IR so that future changes that might break this would result in a change in this test which hopefully someone would catch

Reply:
no_implicit_float can be caught with some IR check and I agree it is better to check the IR.
But the early return due to no vector registers can't, so using a debug statement seems to be the only way to check it.

Reply:
why can't the vector register check be tested the same way?

Reply:
because if the target contains no vector registers the code won't get vectorized because the TTI won't let you get a register wide enough to fit 2 or more elements. This check will just save some compile time.

Reply:
oh I see. then we don't need a test for it if it'll be unobservable anyway

Reply:
OK let me drop this test and replace the other one with an IR snippet that is trivially vectorizable.